code/game/turfs/open/_open.dm

```diff
@@ -107,6 +113,52 @@
 	. = ..()
 	update_visuals()
 
+//ATTACK GHOST IGNORING PARENT RETURN VALUE
+/turf/open/attack_ghost(mob/dead/observer/user)
+	if(destination_z)
+		var/turf/T = locate(destination_x, destination_y, destination_z)
+		user.forceMove(T)
+
+/turf/open/proc/CanBuildHere()
+	if(destination_z)
+		return FALSE
+	return TRUE
```

Comment:
why define this like this? shouldn't we at least be using it to prevent building on mirage'd turfs? does this work ok in gameplay?

Reply:
It DOES prevent building on mirage'd turfs. Don't ask me why, but anytime we check for mirages we check for destination_z. And it works in gameplay. I can't build on it with an RCD and I can't move construction materials there without them being transited.

I don't doubt we'll get some funny issue reports at some point, but there's nothing egregious I can think of